TASK: Bring HTTP docs up to date with actual implementation (#1986)
neos/flow-development-collection: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst

* by default, the :abbr:`HTTP Request Handler (\\Neos\\Flow\\Http\\RequestHandler)` takes over and runs a boot sequence
  which initializes all important parts of Flow
* the HTTP Request Handler builds an PSR-7 HTTP Request and Response object. The

>   :abbr:`Request object (\\Psr\\Http\\Request)` contains all important properties of the real HTTP request.

Author: ```suggestion
  :abbr:`Request object (\\Psr\\Http\\Message\\ServerRequestInterface)` contains all important properties of the real HTTP request.
```

>   The :abbr:`Response object (\\Neos\\Flow\\Http\\Response)` in turn is empty and will be filled with information by a

Author: ```suggestion
  The :abbr:`Response object (\\Psr\Http\\Message\\ResponseInterface)` in turn is empty and will be filled with information by a
```

Request
-------

In the PSR-7 specification, a distinction is made between two different types of requests - incoming (``ServerRequest``)

> and outgoing (``Request``). Whenever you want to make an outgoing request, you can easily use for example Guzzle's
> ``Request`` class constructor with the respective arguments for method, uri, etc. and then pass that to e.g. a PSR-18

Reviewer: tiny wording improvement
```suggestion
and outgoing (``Request``). Whenever you want to make an outgoing request, you can easily use the Guzzle
``Request`` class constructor for example with the respective arguments for method, uri, etc. and then pass that to e.g. a PSR-18
```

> Http Client implementation.

Author: We need to make our own `Browser` PSR-18 compatible at some point. Right now it still uses a `ServerRequest` though in order to provide easy request forwarding in functional tests.

On the other side the incoming request is something you should never try to create an instance of yourself, as it is

> provided by the framework. In theory, you could also call inject the ``ServerRequestFactory::createServerRequest`` or

Reviewer: ```suggestion
provided by the framework. In theory, you could also call the ``ServerRequestFactory::createServerRequest`` or
```

> Guzzle's ``ServerRequest::fromGlobals()`` convenience method, but this does not have any relation to the current request

Reviewer: ```suggestion
the Guzzle ``ServerRequest::fromGlobals()`` convenience method, but this does not have any relation to the current request
```

object handled by the framework. It will not have any of the processing from components applied and might therefore lead
to unexpected results, like the trusted proxy headers ``X-Forwarded-*`` not being applied and the ``ServerRequest`` providing
wrong protocol, host or client IP address.

> If you need access to the **current** HTTP ``Request``, either create a Http Component or only access it inside the

Author: This is in preparation of the deprecation of `ComponentContext` with PSR-15

> If you need access to the **current** HTTP ``Request``, either create a Http Component or only access it inside the

Author: ```suggestion
If you need access to the **current** HTTP ``Request``, either create a :ref:`Http Component<Component Chain>` or only access it inside the
```

> controller through the ``ActionRequest`` for inspecting.

Reviewer: A link to the HTTP Component section and a mini example for the ActionRequest case would be rad

> controller through the ``ActionRequest`` for inspecting.

Author: ```suggestion
controller through the ``ActionRequest`` for inspecting::

	public function myAction() {
		$requestBody = $this->request->getHttpRequest()->getParsedBody();
		...
	}
```
Not sure that's what you were aiming for?

Reviewer: Yes, great, thanks :)

    $requestHandler = $this->bootstrap->getActiveRequestHandler();
    if ($requestHandler instanceof HttpRequestHandlerInterface) {
        $actionRequest = ActionRequest::fromHttpRequest($requestHandler->getHttpRequest());
        // ...
    }

Arguments
~~~~~~~~~

The ``ActionRequest`` features a few methods for retrieving and setting arguments. These arguments are the result of merging any
GET, POST and PUT arguments and even the information about uploaded files. Note that these arguments have already been processed
by the validation and property mapping layerns and thus are suitable for being used in controller actions. If you, however, need to

> access the raw data, you can access these via the ``HttpRequest`` ``getCookieParams()``, ``getQueryParams()``, ``getUploadedFiles()``
> and ``getParsedBody()`` methods respectively.

Reviewer: ```suggestion
access the raw data, you can access these via the ``getCookieParams()``, ``getQueryParams()``, ``getUploadedFiles()`` and ``getParsedBody()``
methods of the ``HttpRequest``  respectively.
```
